pkg/replication/replication.go

```diff
@@ -221,6 +225,12 @@ func (r *replication) Enact() {
 				exitCh := make(chan struct{})
 				timeoutCh := make(chan struct{})
 
+				// Wait until we can read off the rate limit channel before
+				// updating the node
+				_, ok := <-r.rateLimiter.C
+				if !ok {
```

Comment:
A `Ticker`'s channel is never closed. The presence of this check suggests otherwise and is misleading.